File: backend/course-service/src/main/java/com/elearning/course/service/LessonService.java
package com.elearning.course.service;

import com.elearning.course.dto.request.LessonRequest;
import com.elearning.course.dto.response.LessonResponse;
import com.elearning.course.security.AuthenticatedUser;

public interface LessonService {
    LessonResponse create(Long sectionId, LessonRequest request, AuthenticatedUser currentUser);
    LessonResponse update(Long lessonId, LessonRequest request, AuthenticatedUser currentUser);
    void delete(Long lessonId, AuthenticatedUser currentUser);
}

File: frontend/src/app/core/services/lesson.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Lesson, LessonRequest } from '../models/course.model';

@Injectable({ providedIn: 'root' })
export class LessonService {
  private http = inject(HttpClient);
  private baseUrl = environment.courseApiUrl;

  create(sectionId: number, request: LessonRequest): Observable<Lesson> {
    return this.http.post<Lesson>(`${this.baseUrl}/sections/${sectionId}/lessons`, request);
  }

  update(lessonId: number, request: LessonRequest): Observable<Lesson> {
    return this.http.put<Lesson>(`${this.baseUrl}/lessons/${lessonId}`, request);
  }

  delete(lessonId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/lessons/${lessonId}`);
  }
}
